src/cdn_definitions/_impl/__init__.py

```diff
@@ -28,9 +28,9 @@ def get_remote_data(url):
     adapter = requests.adapters.HTTPAdapter(max_retries=retry_strategy)
     session = requests.Session()
     session.mount(url, adapter)
```

Comment:
Note, if changing this to accept a session as input, then we should remove this because it would be an unexpected side effect that a new adapter is mounted onto the passed session.

May I suggest instead either:

- using a plain old retry loop here, or
- just make it so that retry is only enabled when cdn_definitions creates the session itself